extensions/trace_propagators/src/test/java/io/opentelemetry/extensions/trace/propagation/OtTracerPropagatorTest.java

```diff
@@ -0,0 +1,283 @@
+    carrier.put(OtTracerPropagator.SAMPLED_HEADER, PropagatorUtils.FALSE_INT);
+
+    assertThat(getSpanContext(propagator.extract(Context.current(), carrier, getter)))
+        .isEqualTo(
+            SpanContext.createFromRemoteParent(
+                SHORT_TRACE_ID, SPAN_ID, TraceFlags.getDefault(), TRACE_STATE_DEFAULT));
+  }
+
+  @Test
+  public void extract_InvalidTraceId() {
+    Map<String, String> invalidHeaders = new LinkedHashMap<>();
+    invalidHeaders.put(OtTracerPropagator.TRACE_ID_HEADER, "abcdefghijklmnopabcdefghijklmnop");
+    invalidHeaders.put(OtTracerPropagator.SPAN_ID_HEADER, SPAN_ID_BASE16);
+    invalidHeaders.put(OtTracerPropagator.SAMPLED_HEADER, PropagatorUtils.TRUE_INT);
+    assertThat(getSpanContext(propagator.extract(Context.current(), invalidHeaders, getter)))
+        .isSameInstanceAs(SpanContext.getInvalid());
+  }
+
+  @Test
+  public void extract_InvalidTraceId_Size() {
+    Map<String, String> invalidHeaders = new LinkedHashMap<>();
+    invalidHeaders.put(OtTracerPropagator.TRACE_ID_HEADER, TRACE_ID_BASE16 + "00");
+    invalidHeaders.put(OtTracerPropagator.SPAN_ID_HEADER, SPAN_ID_BASE16);
+    invalidHeaders.put(OtTracerPropagator.SAMPLED_HEADER, PropagatorUtils.TRUE_INT);
+    assertThat(getSpanContext(propagator.extract(Context.current(), invalidHeaders, getter)))
+        .isSameInstanceAs(SpanContext.getInvalid());
+  }
+
+  @Test
+  public void extract_InvalidSpanId() {
+    Map<String, String> invalidHeaders = new LinkedHashMap<>();
+    invalidHeaders.put(OtTracerPropagator.TRACE_ID_HEADER, TRACE_ID_BASE16);
+    invalidHeaders.put(OtTracerPropagator.SPAN_ID_HEADER, "abcdefghijklmnop");
+    invalidHeaders.put(OtTracerPropagator.SAMPLED_HEADER, PropagatorUtils.TRUE_INT);
+    assertThat(getSpanContext(propagator.extract(Context.current(), invalidHeaders, getter)))
+        .isSameInstanceAs(SpanContext.getInvalid());
+  }
+
+  @Test
+  public void extract_InvalidSpanId_Size() {
+    Map<String, String> invalidHeaders = new LinkedHashMap<>();
+    invalidHeaders.put(OtTracerPropagator.TRACE_ID_HEADER, TRACE_ID_BASE16);
+    invalidHeaders.put(OtTracerPropagator.SPAN_ID_HEADER, "abcdefghijklmnop" + "00");
+    invalidHeaders.put(OtTracerPropagator.SAMPLED_HEADER, PropagatorUtils.TRUE_INT);
+    assertThat(getSpanContext(propagator.extract(Context.current(), invalidHeaders, getter)))
+        .isSameInstanceAs(SpanContext.getInvalid());
+  }
+
+  @Test
+  public void fieldsList() {
+    assertThat(propagator.fields())
+        .containsExactly(
+            OtTracerPropagator.TRACE_ID_HEADER,
+            OtTracerPropagator.SPAN_ID_HEADER,
+            OtTracerPropagator.SAMPLED_HEADER);
+  }
+
+  @Test
+  public void headerNames() {
+    assertThat(OtTracerPropagator.TRACE_ID_HEADER).isEqualTo("ot-tracer-traceid");
+    assertThat(OtTracerPropagator.SPAN_ID_HEADER).isEqualTo("ot-tracer-spanid");
+    assertThat(OtTracerPropagator.SAMPLED_HEADER).isEqualTo("ot-tracer-sampled");
+  }
+
```

Comment:
```suggestion
```

These tests are basically equivalent to the code, just checking constants, don't think they provide value.